src/Gaffer/Expression.cpp

```diff
@@ -423,7 +423,7 @@ void Expression::updatePlug( ValuePlug *parentPlug, size_t childIndex, ValuePlug
 	// Finally we can add the plug we need.
 
 	PlugPtr childPlug = plug->createCounterpart( "p0", parentPlug->direction() );
-	childPlug->setFlags( Plug::Dynamic, true );
+	childPlug->setFlags( Plug::Dynamic | Plug::Serialisable, true );
```

Comment:
Is this what prevented the serialisation when saving the scene while dispatching? And the rest of the PR is additional improvements, or it's all related?

Reply:
The [ValuePlug fix](36a26bb) is the crucial one. This was a separate one I stumbled on while testing. This one would prevent you from reading from `ImageReader.out.format` in an expression, but I presume you've been working around that by reading from `NodeBelowTheImageReader.in.format`...